include/alpaka/math/abs/Traits.hpp

```diff
@@ -44,7 +54,7 @@ namespace alpaka
         ALPAKA_FN_HOST_ACC auto abs(T const& abs_ctx, TArg const& arg)
         {
             using ImplementationBase = concepts::ImplementationBase<ConceptMathAbs, T>;
-            return traits::Abs<ImplementationBase, TArg>::abs(abs_ctx, arg);
+            return traits::Abs<ImplementationBase, TArg>{}(abs_ctx, arg);
```

Comment:
I think i don't fully understand it. Now a user calls that `alpaka::abs(acc, arg)` declared on line 54 in their kernel. It in turn does ADL on line 40 to switch to user's `abs()` definition when applicable. But when and how do we switch to backend-specific calls such as `struct Abs<AbsUniformCudaHipBuiltIn, TArg, std::enable_if_t<std::is_floating_point<TArg>::value>>` ?

Reply:
I think the schema of how these things are organized is worth documenting. As seen from my question, it's probably not that simple to graps, even though I discovered lack of ADL there and tried a simple fix (with a different way than yours)

Reply:
The user calls `alpaka::math::abs(acc, value);` and the function on line 54 is picked. On line 57 we instantiate `traits::Abs<ImplementationBase, TArg>`. Notice that the template arguments are the backend and the type on which to compute the abs. If there is any specialization of `traits::Abs`, it will be picked. For `float` and `double`, such specializations exist for all backends. However, if there is no specialization, the primary template on line 33 will be instantiated which proceeds to do an ADL call.

We could document that somewhere. Where could be a good place?

Reply:
Thanks for explaining @bernhardmgruber . Where to document is a good question. I am not sure, maybe as a new subsection [here](https://alpaka.readthedocs.io/en/latest/dev/details.html). Btw I've also found out [the last point about naming here](https://alpaka.readthedocs.io/en/latest/dev/style.html#traits) is violated by this PR. However, I remember Benjamin himself recommending this new naming you used, so should be no issue there.

Reply:
As you said, @BenjaminW3 mentioned in some comment somewhere, that we should have picked `operator()` for trait implementations.
Furthermore, I had to change the function name to make it work. Consider this:
```c++
struct Acos
{
    ALPAKA_FN_HOST_ACC auto acos(T const& ctx, TArg const& arg) // instead of operator()
    {
        alpaka::ignore_unused(ctx);
        // This is an ADL call. If you get a compile error here then your type is not supported by the
        // backend and we could not find acos(TArg) in the namespace of your type.
        return acos(arg);
    }
};
```
<details>
  <summary>Guess what would happen :)</summary>
  An infinite recursion :D
</details>